### PR TITLE
Nexus 4635 nexus version

### PR DESCRIPTION
**Pull request by velo**
Based on tamas tests results applied some changes to nexusVersion
https://issues.sonatype.org/browse/NEXUS-4635?focusedCommentId=144699&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-144699
- Included default nexusVersion on nexus.xml
- Fixed loadConfiguration() before everytime loadCfg() was invoked it set isConfigurationUpgraded to false

Also I placed some ITs

### CODE REVIEW

File: nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4635nexusVersion/Nexus4635OldNexusVersionIT.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import org.sonatype.nexus.configuration.application.NexusConfiguration;
+import org.sonatype.nexus.integrationtests.AbstractNexusIntegrationTest;
+import org.sonatype.nexus.rest.model.StatusResource;
+import org.sonatype.nexus.test.utils.NexusConfigUtil;
+import org.testng.annotations.Test;
+
+/**
+ * tampering the nexusVersion to 1.9.3<BR>
+ * http://goo.gl/dxQbh
```
Comment (nabcos) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4635nexusVersion/Nexus4635OldNexusVersionIT.java:32]: I would prefer quote, summarized content or long URL here.

File: nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4635upgrade/Nexus4635FullUpgradeIT.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+{
+
+    public NexusConfigUtil getNexusConfigUtil()
+    {
+        // prevent IT from creating nexus.xml
+        return new NexusConfigUtil( this )
+        {
+            @Override
+            public void validateConfig()
+                throws Exception
```
Comment (nabcos) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4635upgrade/Nexus4635FullUpgradeIT.java]: Would `setVerifyNexusConfigBeforeStart(false)` suffice? easier to comprehend IMO

**Comment by nabcos** [comment]
+1

File: nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4635nexusVersion/Nexus4635OldNexusVersionIT.java (hunk 1 above)
Comment (velo) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4635nexusVersion/Nexus4635OldNexusVersionIT.java:32]: Well, I did add the relevant content here. But used a short URL cause the original long one would be split it 2 lines when code is formatted.

File: nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4635upgrade/Nexus4635FullUpgradeIT.java (hunk 2 above)
Comment (velo) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4635upgrade/Nexus4635FullUpgradeIT.java]: Done!

**Comment by velo** [comment]
Also this pull has a complementary change on PRO side
https://github.com/sonatype/nexus-enterprise/pull/40

File: nexus/nexus-oss-edition/src/main/filtered-resources/META-INF/nexus/nexus.xml [hunk 3]
```diff
@@ -21,6 +21,7 @@
 -->
 <nexusConfiguration>
   <version>1.10.0</version>
+  <nexusVersion>${project.version}</nexusVersion>
```
Comment (cstamas) [inline nexus/nexus-oss-edition/src/main/filtered-resources/META-INF/nexus/nexus.xml:24]: I think this is good but not enough.

https://github.com/sonatype/nexus/blob/master/nexus/nexus-configuration-model/src/main/resources/META-INF/nexus/default-oss-nexus.xml

One more to go.

**Comment by cstamas** [comment]
After you align (just make the same) the two nexus.xmls, is a +1

**Pull request merged**